service-clients-core/src/main/java/com/flipkart/poseidon/serviceclients/ServiceResponseDecoder.java

```diff
@@ -101,9 +101,9 @@ public ServiceResponseDecoder(ObjectMapper objectMapper, JavaType javaType, Java
                         errorResponse = objectMapper.readValue(serviceResponse, errorType);
                     }
                 } catch (Exception e) {
-                    logger.warn("Error while de-serializing non 200 response to given errorType", e);
+                    logger.warn("Error while de-serializing non 200 response to given errorType statusCode:{} exception: {}", statusCodeString, e.getMessage());
```

Comment:
This should be kept as exception, since its a exception we might want to look at